### PR TITLE
make egressfirewall use logical_router_policies

### PR DESCRIPTION
**Pull request by JacobTanenbaum**
Change egressfirewall to use logical_router_policies on the
ovn_cluster_router as oppsed to using ACLs on each nodes join
switches.

This changes optimizes the generation of egressfirewall policies
because we only have to create one object and put it in one place
the ovn_cluster_router applies to all namespaces so we only need
to attach the policy there.

because the single router applies to all namespaces the test
    "correctly adds an existing egressFirewall to a new node" is no longer needed

Signed-off-by: Jacob Tanenbaum <jtanenba@redhat.com>

fixes https://bugzilla.redhat.com/show_bug.cgi?id=1879908

because of https://github.com/ovn-org/ovn-kubernetes/pull/1692 egress traffic does not route through the nodes join switch and the ovn_cluster_router provides a more efficient way to accomplish the same result 

**- What this PR does and why is it needed**

**- Special notes for reviewers**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: go-controller/pkg/ovn/egressfirewall.go [hunk 1]
```diff
@@ -188,30 +170,17 @@ func (ef *egressFirewall) addACLToJoinSwitch(joinSwitches []string, hashedAddres
 		if len(rule.ports) > 0 {
 			match = fmt.Sprintf("%s && ( %s )\"", match[:len(match)-1], egressGetL4Match(rule.ports))
 		}
-		uuid, stderr, err := util.RunOVNNbctl("--data=bare", "--no-heading",
-			"--columns=_uuid", "find", "ACL", match, "action="+action,
-			fmt.Sprintf("external-ids:egressFirewall=%s", ef.namespace))
 
+		_, stderr, err := util.RunOVNNbctl("--id=@logical_router_policy", "create", "logical_router_policy",
+			fmt.Sprintf("priority=%d", defaultStartPriority-rule.id),
```
Comment (dcbw) [inline go-controller/pkg/ovn/egressfirewall.go:175]: @JacobTanenbaum now that our range is smaller, do we have a problem with number of rules? Or do we cap at 99 or something?

**Comment by dcbw** [comment]
Much better approach, if only we'd thought of the router policies earlier (or maybe they weren't in OVN yet, I forget...)

File: go-controller/pkg/ovn/egressfirewall.go (hunk 1 above)
Comment (JacobTanenbaum) [reply go-controller/pkg/ovn/egressfirewall.go:175]: so we can cap the supported number of rules to 99 but that is also the number of rules in a namespace, not for the system as a whole and currently there is a limit in openshift sdn of 50 so it is higher then the sdn version. Lastly if we ever run out of space we can raise all the other priorities to give the firewall more room at the base.

Comment (dcbw) [reply go-controller/pkg/ovn/egressfirewall.go:179]: Note that Dan Winship filed a PR for SDN to increase that number to 1000 or something, so this is no longer a good rationale.

File: go-controller/pkg/ovn/egressfirewall.go [hunk 2]
```diff
@@ -188,30 +174,17 @@ func (ef *egressFirewall) addACLToJoinSwitch(joinSwitches []string, hashedAddres
...
-				if err != nil {
-					return fmt.Errorf("error executing create ACL command, stderr: %q, %+v", stderr, err)
-				}
-			} else {
-				_, stderr, err := util.RunOVNNbctl("add", "logical_switch", joinSwitch, "acls", uuid)
-				if err != nil {
-					return fmt.Errorf("error adding ACL to joinsSwitch %s failed, stderr: %q, %+v", joinSwitch, stderr, err)
-
-				}
+			// TODO: lr-policy-add doesn't support --may-exist, resort to this workaround for now.
```
Comment (dcbw) [inline go-controller/pkg/ovn/egressfirewall.go:183]: @JacobTanenbaum can you file a RH bugzilla bug against OVN for this?

Comment (trozet) [reply go-controller/pkg/ovn/egressfirewall.go:183]: I already filed one a while ago
https://bugzilla.redhat.com/show_bug.cgi?id=1845109

File: go-controller/pkg/ovn/egressfirewall.go [hunk 3]
```diff
@@ -15,7 +15,7 @@ import (
 )
 
 const (
-	defaultStartPriority           = 2000
+	defaultStartPriority           = 100
 	egressFirewallAppliedCorrectly = "EgressFirewall Rules applied"
```
Comment (trozet) [inline go-controller/pkg/ovn/egressfirewall.go:19]: we probably need to discuss the priorities, ping me on Monday @JacobTanenbaum

**Pull request merged**